packages/explorerkit-server/src/components/idls.ts

```diff
@@ -9,15 +9,20 @@ const IDL_CACHE_TTL = 3600; // one hour
 
 export async function loadAllIdls(programIds: string[]): Promise<IdlsMap> {
   const idls: IdlsMap = new Map();
+
+  if (programIds.length === 0) {
+    return idls;
+  }
+
   const cache = getSharedDep("cache");
-  const cachedIdls = await Promise.allSettled(programIds.map((id) => cache.get(id)));
+  const cachedIdls = await cache.multiGet(programIds);
 
   await Promise.allSettled(
     cachedIdls.map(async (res, i) => {
       const programId = programIds[i]!;
```

Comment:
It is mostly because missing idls need to make an RPC call to solana-fm to retrieve and store that in the cache. See line below:
```ts
const idl = await getProgramIdl(programId);
```